### DISCUSSION TITLE
Mocking Prisma Using Jest

### DISCUSSION THREAD

**Discussion by rtnolan** [prisma/prisma#3275] [Q&A] [answered]
Hi guys,

I'm currently using `Jest` & `jest-mock-extended` in order to mock the creation of some instances for testing purposes. I have two models which look like: 

```prisma
model Strategy {
  id String @default(cuid()) @id
  name String @unique
  market StrategyMarketComposition[]
}

model StrategyMarketComposition {
  id String @default(cuid()) @id
  region String?
  country String?
  strategy    Strategy  @relation(fields: [strategyId], references: [id])
  strategyId  String
}
```
Right now I'm able to mock the creation of a Strategy easily enough using:
```javascript
const prisma = mockDeep<PrismaClient>();

describe("Create Strategy", () => {
  test("Create Strategy", async (done) => {
    prisma.strategy.create.mockResolvedValue({
      id: "1",
      name: "My Strategy"
    });

    service.createStrategy(input).then((data) => {
      if (data) {
        expect(data.name).toStrictEqual("My Strategy");
      }
   });
   done();
}
```

Where I'm struggling is trying to create a `StrategyMarketComposition` within the mock. I've tried something like: 
```javascript
prisma.strategy.create.mockResolvedValue({
  id: "1",
  name: "My Strategy"
  market: prisma.strategyMarketComposition.create.mockResolvedValue([{
    id: "1",
    region: "A",
    country: "AA,
    strategyId: "1"
  }]
});
```
But nothing I do seems to agree with the params it expects. Any advice on how I might be able to mock the creation of another model within the same mock? Any help is really appreciated!

\- Tom

**Comment by ryands17** [comment, marked as answer]
Hey @tnolan8 👋 
There is a discussion comment https://github.com/prisma/prisma/discussions/2083#discussion-901 that explains how to use this. I personally haven't used this and rely more on Integration tests with an actual DB rather than mocking Prisma itself. If you wish to view those, you can find that [here](https://github.com/prisma/prisma-examples/tree/master/typescript/testing-express) and [here](https://github.com/ryands17/nexus-auth/).

Let me know if the above works.

**Reply by rtnolan** [reply]
Hey @ryands17 👋 
Yeah that's what I was following along with but it doesn't show how to mock another instance within creating the first if that makes sense. I may look to switch up our testing and go down the route of using a database instead, thank you.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
